--- cacti-1.2.26/include/global.php~	2023-12-24 03:17:14.000000000 +0100
+++ cacti-1.2.26/include/global.php	2024-02-28 13:34:23.293235210 +0100
@@ -81,13 +81,7 @@ $disable_log_rotation = false;
 
 $config = array();
 
-/* Include configuration, or use the defaults */
-if (file_exists(dirname(__FILE__) . '/config.php')) {
-	if (!is_readable(dirname(__FILE__) . '/config.php')) {
-		die('Configuration file include/config.php is present, but unreadable.' . PHP_EOL);
-	}
-	include(dirname(__FILE__) . '/config.php');
-}
+require '/etc/webapps/cacti/config.php';
 
 if (isset($config['cacti_version'])) {
 	die('Invalid include/config.php file detected.' . PHP_EOL);
@@ -238,7 +232,7 @@ if ($config['cacti_server_os'] == 'win32
 }
 
 $config['include_path'] = dirname(__FILE__);
-$config['rra_path'] = $config['base_path'] . '/rra';
+$config["rra_path"] = '/var/lib/cacti/rra';
 
 /* for multiple pollers, we need to know this location */
 if (!isset($scripts_path)) {
--- cacti-1.2.26/include/config.php~	2023-12-24 03:17:14.000000000 +0100
+++ cacti-1.2.26/include/config.php	2024-02-28 13:35:19.813235204 +0100
@@ -29,8 +29,8 @@
 $database_type     = 'mysql';
 $database_default  = 'cacti';
 $database_hostname = 'localhost';
-$database_username = 'cactiuser';
-$database_password = 'cactiuser';
+$database_username = 'mysql';
+$database_password = '';
 $database_port     = '3306';
 $database_retries  = 5;
 $database_ssl      = false;
--- cacti-1.2.26/poller.php~	2023-12-24 03:17:14.000000000 +0100
+++ cacti-1.2.26/poller.php	2024-02-28 13:36:02.066568538 +0100
@@ -31,7 +31,7 @@ if (function_exists('pcntl_async_signals
 
 ini_set('output_buffering', 'Off');
 
-require(__DIR__ . '/include/cli_check.php');
+require('/usr/share/cacti/include/cli_check.php');
 require_once($config['base_path'] . '/lib/poller.php');
 require_once($config['base_path'] . '/lib/data_query.php');
 require_once($config['base_path'] . '/lib/rrd.php');
diff -ur cacti-1.2.6.org/install/functions.php cacti-1.2.6/install/functions.php
--- cacti-1.2.6.org/install/functions.php	2019-09-02 10:23:43.000000000 +0200
+++ cacti-1.2.6/install/functions.php	2019-09-26 11:19:59.780907049 +0200
@@ -794,8 +794,8 @@
 			$section = 'general';
 		}
 		$logfile = 'install' . '-' . $section;
-		file_put_contents($config['base_path'] . '/log/' . $logfile . '.log', sprintf($format_log1, $day, $time, $levelname, $data, PHP_EOL), $flags);
-		file_put_contents($config['base_path'] . '/log/install-complete.log', sprintf($format_log2, $day, $time, $sectionname, $levelname, $data, PHP_EOL), $flags);
+		file_put_contents('/var/log/cacti/' . $logfile . '.log', sprintf($format_log1, $day, $time, $levelname, $data, PHP_EOL), $flags);
+		file_put_contents('/var/log/cacti/install-complete.log', sprintf($format_log2, $day, $time, $sectionname, $levelname, $data, PHP_EOL), $flags);
 	}
 }
 
--- cacti-1.2.26/lib/installer.php~	2023-12-24 03:17:14.000000000 +0100
+++ cacti-1.2.26/lib/installer.php	2024-02-28 13:37:29.589901867 +0100
@@ -528,7 +528,6 @@ class Installer implements JsonSerializa
 
 		$always_paths = array(
 			'sys_temp'  => sys_get_temp_dir(),
-			'log'       => $config['base_path'] . '/log',
 			'boost'     => $config['base_path'] . '/cache/boost',
 			'mibcache'  => $config['base_path'] . '/cache/mibcache',
 			'realtime'  => $config['base_path'] . '/cache/realtime',
@@ -3539,7 +3538,7 @@ class Installer implements JsonSerializa
 
 		$page_nr = 1;
 		$total_rows = 500;
-		$logcontents = tail_file($config['base_path'] . '/log/cacti.log', 100, -1, ' INSTALL:' , $page_nr, $total_rows);
+                $logcontents = tail_file('/var/log/cacti/cacti.log', 100, -1, ' INSTALL:' , $page_nr, $total_rows);
 
 		$output_log = '';
 		foreach ($logcontents as $logline) {

