diff -urN cacti-0.8.7a.orig/include/config.php cacti-0.8.7a/include/config.php
--- cacti-0.8.7a.orig/include/config.php	2007-11-17 19:11:52.000000000 +0100
+++ cacti-0.8.7a/include/config.php	2007-12-04 10:53:28.000000000 +0100
@@ -30,6 +30,8 @@
 $database_password = "cactiuser";
 $database_port = "3306";
 
+include '/etc/cacti/cacti.cfg';
+
 /* Default session name - Session name must contain alpha characters */
 #$cacti_session_name = "Cacti";
 
