--- cacti-0.8.7b/include/global.php	2008-10-05 04:38:29.740276226 +0300
+++ cacti-0.8.7g/include/global.php	2010-12-13 12:10:44.312310245 +0200
@@ -42,7 +42,7 @@
 $cacti_session_name = "Cacti";
 
 /* Include configuration */
-include(dirname(__FILE__) . "/config.php");
+require '/etc/webapps/cacti/config.php';
 
 if (isset($config["cacti_version"])) {
 	die("Invalid include/config.php file detected.");
@@ -101,7 +101,8 @@
 $config["base_path"] = strtr(ereg_replace("(.*)[\/\\]include", "\\1", dirname(__FILE__)), "\\", "/");
 $config["library_path"] = ereg_replace("(.*[\/\\])include", "\\1lib", dirname(__FILE__));
 $config["include_path"] = dirname(__FILE__);
-$config["rra_path"] = $config["base_path"] . '/rra';
+$config["rra_path"] = '/var/lib/cacti/rra';
+$config["log_path"] = '/var/log/cacti';
 
 /* colors */
 $colors["dark_outline"] = "454E53";
--- cacti-0.8.7g/include/config.php~	2010-12-13 12:15:08.000000000 +0200
+++ cacti-0.8.7g/include/config.php	2010-12-13 12:16:09.275611204 +0200
@@ -26,8 +26,8 @@
 $database_type = "mysql";
 $database_default = "cacti";
 $database_hostname = "localhost";
-$database_username = "cactiuser";
-$database_password = "cactiuser";
+$database_username = "mysql";
+$database_password = "";
 $database_port = "3306";
 
 /* load up old style plugins here */
@@ -34,4 +34,6 @@
 /* Default session name - Session name must contain alpha characters */
 #$cacti_session_name = "Cacti";
 
+$url_path = '/cacti/';
+
 ?>
--- cacti-0.8.7b/include/global_settings.php~	2008-10-05 04:29:33.121404239 +0300
+++ cacti-0.8.7b/include/global_settings.php	2008-10-05 04:41:11.807395633 +0300
@@ -96,7 +96,7 @@
 			"friendly_name" => "Cacti Log File Path",
 			"description" => "The path to your Cacti log file (if blank, defaults to <path_cacti>/log/cacti.log)",
 			"method" => "filepath",
-			"default" => $config["base_path"] . "/log/cacti.log",
+			"default" => $config["log_path"] . "/cacti.log",
 			"max_length" => "255"
 			),
 		"pollerpaths_header" => array(
--- cacti-0.8.7b/install/index.php~	2008-10-05 04:29:33.113733438 +0300
+++ cacti-0.8.7b/install/index.php	2008-10-05 04:41:37.276839747 +0300
@@ -244,7 +244,7 @@
 if (config_value_exists("path_cactilog")) {
 	$input["path_cactilog"]["default"] = read_config_option("path_cactilog");
 } else {
-	$input["path_cactilog"]["default"] = $config["base_path"] . "/log/cacti.log";
+	$input["path_cactilog"]["default"] = $config["log_path"] . "/cacti.log";
 }
 
 /* SNMP Version */
--- cacti-0.8.7b/lib/functions.php~	2008-10-05 04:29:33.124737562 +0300
+++ cacti-0.8.7b/lib/functions.php	2008-10-05 04:42:05.046758504 +0300
@@ -367,7 +367,7 @@
 	/* Log to Logfile */
 	if ((($logdestination == 1) || ($logdestination == 2)) && (read_config_option("log_verbosity") != POLLER_VERBOSITY_NONE)) {
 		if ($logfile == "") {
-			$logfile = $config["base_path"] . "/log/cacti.log";
+			$logfile = $config["log_path"] . "/cacti.log";
 		}
 
 		/* echo the data to the log (append) */
--- cacti-0.8.8/utilities.php~	2012-10-29 22:14:05.000000000 +0200
+++ cacti-0.8.8/utilities.php	2012-10-29 22:15:05.781005237 +0200
@@ -798,12 +798,12 @@
 }
 
 function utilities_view_logfile() {
-	global $colors, $log_tail_lines, $page_refresh_interval, $refresh;
+	global $colors, $log_tail_lines, $page_refresh_interval, $config;
 
 	$logfile = read_config_option("path_cactilog");
 
 	if ($logfile == "") {
-		$logfile = "./log/rrd.log";
+		$logfile = $config['log_path']. "/rrd.log";
 	}
 
 	/* helps determine output color */
@@ -858,7 +858,7 @@
 }
 
 function utilities_clear_logfile() {
-	global $colors;
+	global $config, $colors;
 
 	load_current_session_value("refresh", "sess_logfile_refresh", read_config_option("log_refresh_interval"));
 
@@ -870,7 +870,7 @@
 	$logfile = read_config_option("path_cactilog");
 
 	if ($logfile == "") {
-		$logfile = "./log/cacti.log";
+		$logfile = $config['log_path']."/cacti.log";
 	}
 
 	html_start_box("<strong>Clear Cacti Log File</strong>", "100%", $colors["header"], "1", "center", "");
--- cacti/poller.php~	2008-10-05 05:08:10.000000000 +0300
+++ cacti/poller.php	2008-10-05 05:11:11.327705168 +0300
@@ -1,3 +1,4 @@
+#!/usr/bin/php
 <?php
 /*
  +-------------------------------------------------------------------------+
@@ -32,7 +32,7 @@
 $no_http_headers = true;
 
 /* start initialization section */
-include(dirname(__FILE__) . "/include/global.php");
+require "/usr/share/cacti/include/global.php";
 include_once($config["base_path"] . "/lib/poller.php");
 include_once($config["base_path"] . "/lib/data_query.php");
 include_once($config["base_path"] . "/lib/graph_export.php");
