--- /poller.php~	2008-10-05 16:54:15.000000000 +0300
+++ /poller.php	2008-10-05 16:57:05.946911539 +0300
@@ -181,7 +181,9 @@
 	$last_host           = 0;
 
 	/* update web paths for the poller */
-	db_execute("replace into settings (name,value) values ('path_webroot','" . addslashes(($config["cacti_server_os"] == "win32") ? strtr(strtolower(substr(dirname(__FILE__), 0, 1)) . substr(dirname(__FILE__), 1),"\\", "/") : dirname(__FILE__)) . "')");
+//	$path_webroot = ($config["cacti_server_os"] == "win32") ? strtr(strtolower(substr(dirname(__FILE__), 0, 1)) . substr(dirname(__FILE__), 1),"\\", "/") : dirname(__FILE__);
+	$path_webroot = $config['base_path'];
+	db_execute("replace into settings (name,value) values ('path_webroot','" .  addslashes($path_webroot) . "')");
 
 	/* obtain some defaults from the database */
 	$poller      = read_config_option("poller_type");
