--- /poller.php~	2008-10-05 16:54:15.000000000 +0300
+++ /poller.php	2008-10-05 16:57:05.946911539 +0300
@@ -424,7 +424,7 @@ while ($poller_runs_completed < $poller_
 	$started_processes = 0;
 	$first_host        = 0;
 	$last_host         = 0;
-	$webroot           = addslashes(($config['cacti_server_os'] == 'win32') ? strtr(strtolower(substr(dirname(__FILE__), 0, 1)) . substr(dirname(__FILE__), 1),"\\", '/') : dirname(__FILE__));
+	$webroot = $config['base_path'];
 
 	// update web paths for the poller
 	set_config_option('path_webroot', $webroot);

