--- ./scripts/ss_host_disk.php	2008-10-05 18:19:17.656634044 +0300
+++ ./scripts/ss_host_disk.php	2008-10-07 11:07:10.120905407 +0300
@@ -86,12 +86,16 @@ function ss_host_disk($hostname, $host_i
 				$snmp_auth_password, $snmp_auth_protocol, $snmp_priv_passphrase, $snmp_priv_protocol,
 				$snmp_context, $snmp_port, $snmp_timeout, $ping_retries, $max_oids, SNMP_POLLER)
 			);
+		if (empty($arr_index))
+			return;
 
 		$arr = ss_host_disk_reindex(
 			cacti_snmp_walk($hostname, $snmp_community, $oids[$arg], $snmp_version, $snmp_auth_username,
 				$snmp_auth_password, $snmp_auth_protocol, $snmp_priv_passphrase, $snmp_priv_protocol,
 				$snmp_context, $snmp_port, $snmp_timeout, $ping_retries, $max_oids, SNMP_POLLER)
 			);
+		if (empty($arr))
+			return; // we likely already got error already earlier, like "Timeout: No Response from localhost:161"
 
 		for ($i=0;($i<sizeof($arr_index));$i++) {
 			print $arr_index[$i] . '!' . $arr[$i] . "\n";

