--- cacti-0.8.7b/poller.php~	2008-10-05 16:58:33.000000000 +0300
+++ cacti-0.8.7b/poller.php	2008-10-05 17:45:53.825952709 +0300
@@ -665,7 +665,8 @@ function log_cacti_stats($loop_start, $m
 	);
 
 	$cacti_stats = vsprintf('Time:%01.4f Method:%s Processes:%s Threads:%s Hosts:%s HostsPerProcess:%s DataSources:%s RRDsProcessed:%s', $perf_data);
-	cacti_log('STATS: ' . $cacti_stats , true, 'SYSTEM');
+	if (read_config_option('log_verbosity') > POLLER_VERBOSITY_NONE)
+		cacti_log('STATS: ' . $cacti_stats , true, 'SYSTEM');
 
 	/* insert poller stats into the settings table */
 	db_execute_prepared('REPLACE INTO settings (name, value) VALUES ("stats_poller",?)', array($cacti_stats));

