--- cacti-1.2.26/poller.php~	2024-02-28 13:37:51.000000000 +0100
+++ cacti-1.2.26/poller.php	2024-02-28 13:38:33.309901864 +0100
@@ -1093,8 +1093,9 @@ function log_cacti_stats($loop_start, $m
 		$rrds_processed
 	);
 
-	$cacti_stats = vsprintf('Time:%01.4f Method:%s Processes:%s Threads:%s Hosts:%s HostsPerProcess:%s DataSources:%s RRDsProcessed:%s', $perf_data);
-	cacti_log('STATS: ' . $cacti_stats , true, 'SYSTEM');
+        $cacti_stats = vsprintf('Time:%01.4f Method:%s Processes:%s Threads:%s Hosts:%s HostsPerProcess:%s DataSources:%s RRDsProcessed:%s', $perf_data);
+        if (read_config_option('log_verbosity') > POLLER_VERBOSITY_NONE)
+	    cacti_log('STATS: ' . $cacti_stats , true, 'SYSTEM');
 
 	// insert poller stats into the settings table
 	if ($poller_id > 1) {
