--- cacti-0.8.7b/include/global.php	2008-10-05 04:38:29.740276226 +0300
+++ cacti-0.8.7g/include/global.php	2010-12-13 12:10:44.312310245 +0200
@@ -64,10 +64,7 @@ $url_path = '/cacti/';
 /* allow upto 5000 items to be selected */
 ini_set('max_input_vars', '5000');
 
-/* Include configuration, or use the defaults */
-if (file_exists(dirname(__FILE__) . '/config.php')) {
-	include(dirname(__FILE__) . '/config.php');
-}
+require '/etc/webapps/cacti/config.php';
 
 if (isset($config['cacti_version'])) {
 	die('Invalid include/config.php file detected.');
@@ -139,7 +139,8 @@ if ($config['cacti_server_os'] == 'win32
 	$config['library_path'] = preg_replace("/(.*[\/])include/", "\\1lib", dirname(__FILE__));
 }
 $config['include_path'] = dirname(__FILE__);
-$config['rra_path'] = $config['base_path'] . '/rra';
+$config["rra_path"] = '/var/lib/cacti/rra';
+$config["path_cactilog"] = '/var/log/cacti/cacti.log';
 
 /* for multiple pollers, we need to know this location */
 if (!isset($scripts_path)) {
--- cacti-0.8.7g/include/config.php~	2010-12-13 12:15:08.000000000 +0200
+++ cacti-0.8.7g/include/config.php	2010-12-13 12:16:09.275611204 +0200
@@ -27,8 +27,8 @@
 $database_type     = 'mysql';
 $database_default  = 'cacti';
 $database_hostname = 'localhost';
-$database_username = 'cactiuser';
-$database_password = 'cactiuser';
+$database_username = 'mysql';
+$database_password = '';
 $database_port     = '3306';
 $database_ssl      = false;
 
--- cacti/poller.php~	2017-08-28 05:48:36.000000000 +0200
+++ cacti/poller.php	2017-08-31 08:39:19.780902180 +0200
@@ -61,7 +61,7 @@ if (!isset($_SERVER['argv'][0]) || isset
 }
 
 /* start initialization section */
-include(dirname(__FILE__) . '/include/global.php');
+require "/usr/share/cacti/include/global.php";
 include_once($config['base_path'] . '/lib/poller.php');
 include_once($config['base_path'] . '/lib/data_query.php');
 include_once($config['base_path'] . '/lib/rrd.php');

