--- cacti-0.8.7b/include/global.php	2008-10-05 04:38:29.740276226 +0300
+++ cacti-0.8.7g/include/global.php	2010-12-13 12:10:44.312310245 +0200
@@ -64,10 +64,7 @@ $url_path = '/cacti/';
 /* allow upto 5000 items to be selected */
 ini_set('max_input_vars', '5000');
 
-/* Include configuration, or use the defaults */
-if (file_exists(dirname(__FILE__) . '/config.php')) {
-	include(dirname(__FILE__) . '/config.php');
-}
+require '/etc/webapps/cacti/config.php';
 
 if (isset($config['cacti_version'])) {
 	die('Invalid include/config.php file detected.');
@@ -139,7 +139,8 @@ if ($config['cacti_server_os'] == 'win32
 	$config['library_path'] = preg_replace("/(.*[\/])include/", "\\1lib", dirname(__FILE__));
 }
 $config['include_path'] = dirname(__FILE__);
-$config['rra_path'] = $config['base_path'] . '/rra';
+$config["rra_path"] = '/var/lib/cacti/rra';
+$config["path_cactilog"] = '/var/log/cacti/cacti.log';
 
 /* for multiple pollers, we need to know this location */
 if (!isset($scripts_path)) {
--- cacti-0.8.7g/include/config.php~	2010-12-13 12:15:08.000000000 +0200
+++ cacti-0.8.7g/include/config.php	2010-12-13 12:16:09.275611204 +0200
@@ -27,8 +27,8 @@
 $database_type     = 'mysql';
 $database_default  = 'cacti';
 $database_hostname = 'localhost';
-$database_username = 'cactiuser';
-$database_password = 'cactiuser';
+$database_username = 'mysql';
+$database_password = '';
 $database_port     = '3306';
 $database_ssl      = false;
 
--- cacti/poller.php~	2017-08-28 05:48:36.000000000 +0200
+++ cacti/poller.php	2017-08-31 08:39:19.780902180 +0200
@@ -26,7 +26,7 @@
 /* tick use required as of PHP 4.3.0 to accomodate signal handling */
 declare(ticks = 1);
 
-require(__DIR__ . '/include/cli_check.php');
+require('/usr/share/cacti/include/cli_check.php');
 require_once($config['base_path'] . '/lib/poller.php');
 require_once($config['base_path'] . '/lib/data_query.php');
 require_once($config['base_path'] . '/lib/rrd.php');

diff -ur cacti-1.2.6.org/install/functions.php cacti-1.2.6/install/functions.php
--- cacti-1.2.6.org/install/functions.php	2019-09-02 10:23:43.000000000 +0200
+++ cacti-1.2.6/install/functions.php	2019-09-26 11:19:59.780907049 +0200
@@ -794,8 +794,8 @@
 			$section = 'general';
 		}
 		$logfile = 'install' . '-' . $section;
-		file_put_contents($config['base_path'] . '/log/' . $logfile . '.log', sprintf($format_log1, $day, $time, $levelname, $data, PHP_EOL), $flags);
-		file_put_contents($config['base_path'] . '/log/install-complete.log', sprintf($format_log2, $day, $time, $sectionname, $levelname, $data, PHP_EOL), $flags);
+		file_put_contents('/var/log/cacti/' . $logfile . '.log', sprintf($format_log1, $day, $time, $levelname, $data, PHP_EOL), $flags);
+		file_put_contents('/var/log/cacti/install-complete.log', sprintf($format_log2, $day, $time, $sectionname, $levelname, $data, PHP_EOL), $flags);
 	}
 }
 
diff -ur cacti-1.2.6.org/lib/installer.php cacti-1.2.6/lib/installer.php
--- cacti-1.2.6.org/lib/installer.php	2019-09-02 10:23:43.000000000 +0200
+++ cacti-1.2.6/lib/installer.php	2019-09-26 11:20:54.749208954 +0200
@@ -465,7 +465,6 @@
 
 		$always_paths = array(
 			sys_get_temp_dir(),
-			$config['base_path'] . '/log',
 			$config['base_path'] . '/cache/boost',
 			$config['base_path'] . '/cache/mibcache',
 			$config['base_path'] . '/cache/realtime',
@@ -3216,7 +3215,7 @@
 		global $config;
 
 		$page_nr = 1;
-		$logcontents = tail_file($config['base_path'] . '/log/cacti.log', 100, -1, ' INSTALL:' , $page_nr, $total_rows);
+		$logcontents = tail_file('/var/log/cacti/cacti.log', 100, -1, ' INSTALL:' , $page_nr, $total_rows);
 
 		$output_log = '';
 		foreach ($logcontents as $logline) {
