--- cacti-0.8.7e/cli/import_template.php	2010-04-14 18:11:37.575631837 +0300
+++ cacti-0.8.7e/cli/import_template.php	2010-04-14 18:07:35.956754241 +0300
@@ -31,8 +31,8 @@
 /* We are not talking to the browser */
 $no_http_headers = true;
 
-include(dirname(__FILE__)."/../include/global.php");
-include_once("../lib/import.php");
+require(dirname(__FILE__)."/../include/global.php");
+require_once($config["base_path"]."/lib/import.php");
 
 /* process calling arguments */
 $parms = $_SERVER["argv"];
