--- cacti-0.8.7d/lib/html_tree.php	2009-02-11 21:55:50.000000000 -0500
+++ cacti-patched/lib/html_tree.php	2009-03-02 20:00:38.000000000 -0500
@@ -1216,7 +1216,7 @@
 
 	/* generate page list */
 	if ($total_rows > get_request_var_request("graphs")) {
-		$url_page_select = get_page_list(get_request_var_request("page"), MAX_DISPLAY_PAGES, get_request_var_request("graphs"), $total_rows, "graph_view.php?action=tree&tree_id=" . $tree_id . "&leaf_id=" . $leaf_id . (isset($_REQUEST["host_group_data"]) ? "&host_group_data=" . get_request_var_request("host_group_data") : "") . "&filter=" . get_request_var_request("filter"));
+		$url_page_select = get_page_list(get_request_var_request("page"), MAX_DISPLAY_PAGES, get_request_var_request("graphs"), $total_rows, "graph_view.php?action=tree&tree_id=" . $tree_id . "&leaf_id=" . $leaf_id . (isset($_REQUEST["host_group_data"]) ? "&host_group_data=" . get_request_var_request("host_group_data") : ""));
 
 		$nav = "<tr bgcolor='#" . $colors["header"] . "'>
 				<td colspan='11'>
