diff -ruBbd cacti-0.8.7b/lib/snmp.php cacti-0.8.7b-patched/lib/snmp.php
--- cacti-0.8.7b/lib/snmp.php	2008-02-11 19:57:48.000000000 -0500
+++ cacti-0.8.7b-patched/lib/snmp.php	2008-02-29 20:53:00.000000000 -0500
@@ -225,6 +225,7 @@
 function cacti_snmp_walk($hostname, $community, $oid, $version, $username, $password, $auth_proto, $priv_pass, $priv_proto, $context, $port = 161, $timeout = 500, $retries = 0, $environ = SNMP_POLLER) {
 	global $config;
 
+	$snmp_auth	= '';
 	$snmp_array = array();
 	$temp_array = array();
 
