diff -ruBbd cacti-0.8.7b/install/index.php cacti-0.8.7b-patched/install/index.php
--- cacti-0.8.7b/install/index.php	2008-02-14 18:58:01.000000000 -0500
+++ cacti-0.8.7b-patched/install/index.php	2008-02-11 19:57:48.000000000 -0500
@@ -27,7 +27,7 @@
/* allow the upgrade script to run for as long as it needs to */
ini_set("max_execution_time", "0");

+$cacti_versions = array("0.8", "0.8.1", "0.8.2", "0.8.2a", "0.8.3", "0.8.3a", "0.8.4", "0.8.5", "0.8.5a", "0.8.6", "0.8.6a", "0.8.6b", "0.8.6c", "0.8.6d", "0.8.6e", "0.8.6f", "0.8.6g", "0.8.6h", "0.8.6i", "0.8.6j", "0.8.6k", "0.8.7", "0.8.7a", "0.8.7b");
-$cacti_versions = array("0.8", "0.8.1", "0.8.2", "0.8.2a", "0.8.3", "0.8.3a", "0.8.4", "0.8.5", "0.8.5a", "0.8.6", "0.8.6a", "0.8.6b", "0.8.6c", "0.8.6d", "0.8.6e", "0.8.6f", "0.8.6g", "0.8.6h", "0.8.6i", "0.8.6j", "0.8.7", "0.8.7a", "0.8.7b");

$old_cacti_version = db_fetch_cell("select cacti from version");
